network/peers/conn_manager.go

```diff
@@ -192,3 +208,57 @@ func scorePeer(peerSubnets records.Subnets, subnetsScores []float64) PeerScore {
 	}
 	return PeerScore(score)
 }
+
+// DisconnectFromBadPeers will disconnect from bad peers according to their GossipSub scores. It returns the number of disconnected peers.
+func (c connManager) DisconnectFromBadPeers(logger *zap.Logger, net libp2pnetwork.Network, allPeers []peer.ID) int {
+
+	disconnectedPeers := 0
+
+	for _, peerID := range allPeers {
+
+		// Check if peer is bad
+		if isBad, gossipSubScore := c.gossipSubScoreIndex.HasBadGossipSubScore(peerID); isBad {
+
+			// Disconnect
+			err := c.disconnect(peerID, net)
+			if err != nil {
+				logger.Error("Couldn't disconnect from bad peer", fields.PeerID(peerID), zap.Float64("GossipSub Score", gossipSubScore))
+			} else {
+				logger.Debug("Disconnecting from bad peer", fields.PeerID(peerID), zap.Float64("GossipSub Score", gossipSubScore))
```

Comment:
snake case log fields
```suggestion
				logger.Debug("disconnecting from bad peer", fields.PeerID(peerID), zap.Float64("gossip_score", gossipSubScore))
```